streams/src/main/java/org/apache/kafka/streams/kstream/internals/graph/TableProcessorNode.java

```diff
@@ -17,19 +17,21 @@
 
 package org.apache.kafka.streams.kstream.internals.graph;
 
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.streams.kstream.internals.KeyValueStoreMaterializer;
 import org.apache.kafka.streams.kstream.internals.MaterializedInternal;
-import org.apache.kafka.streams.processor.StateStore;
 import org.apache.kafka.streams.processor.internals.InternalTopologyBuilder;
+import org.apache.kafka.streams.state.KeyValueStore;
 
-public class TableProcessorNode<K, V, S extends StateStore> extends StreamsGraphNode {
+public class TableProcessorNode<K, V> extends StreamsGraphNode {
```

Comment:
Why not extend `StatefulProcessorNode` ? Why remove the generic type? Can't it be windowed, too? What about custom stores in `transform()` ?

Reply:
+1.

Reply:
I removed the generic type from the class definition and changed the `materializedInternal` from `S` to `KeyValueStore<Bytes, byte[]` as it matches the `MaterializedInternal` parameter used in `KTable#filter`, `KTable#mapValues`, `KTable#transformValues`.  I can revert that if you want.

Looking at `StatefulProcessorNode` it's only used from `KStream` for `process` and `transform`, but those methods never pass a strore builder or materialized, just store names.

 What makes sense to me is to refactor `StatefulProcessorNode` to remove the store builder then have `TableProcessorNode` extend `StatefulProcessorNode` . WDYT?

I'm not sure what you mean custom stores in `transform()`, `KStream#transform` takes a list of store names which is captured in the `StatefulProcessorNode` is that what you are referring to?

Reply:
EDIT: I've put the generic type back, but requires a cast internally, you have to let me know what you think

Reply:
Ack. Makes sense that `transform` does not apply here as this is a `*Table*ProcessorNode`.